opm/autodiff/NewtonIterationBlackoilCPR.hpp

```diff
@@ -58,12 +70,46 @@ namespace Opm
 
         /// \copydoc NewtonIterationBlackoilInterface::iterations
         virtual int iterations () const { return iterations_; }
+
+        /// \copydoc NewtonIterationBlackoilInterface::parallelInformation
+        virtual const boost::any& parallelInformation() const;
+
     private:
+
+        /// \brief construct the CPR preconditioner and the solver.
+        /// \tparam P The type of the parallel information.
+        /// \param parallelInformation the information about the parallelization.
+        template<int category=Dune::SolverCategory::sequential, class O, class P>   
```

Comment:
Because it is basically two methods:

1. A sequential one:
   O: MatrixtAdapter
   P: SequentialInformation
2. A parallel one:
   O: OverlappingSchwarzOperator
   P: ParallelInformation